### DISCUSSION TITLE
Get a controls absolute position

### DISCUSSION THREAD

**Discussion by Varmod** [AvaloniaUI/Avalonia#12972] [Q&A]
Hi,

I'm looking for a way to retrieve the position of a control in regards to the main window/ main view (top-left corner).
I've looked into the `Bounds` property but it returns the relative position in regards to the parent element.
Is there way to get the absolute position?

**Comment by timunie** [comment]
In worst case you have to traverse the visual tree and sum up the positions. But I wonder what is the usecase here? Why do you need that position?

**Reply by Varmod** [reply]
I'm creating a control that works similar to a popup. I need to pass it information on where it should place it self in regards to a control.

I'll implement it through the tree traversal as that was also my backup plan in case I couldn't find an easier option.

**Reply by timunie** [reply]
you can also look into what popup does. We have overlay popups that also have to calculate their position somehow. 

Still I suggest to look into flyouts. These are really poweful and can do what you need maybe already out of the box.

**Reply by maxkatz6** [reply]
Overlay popups are positioned in the Canvas control that is rendered on top of everything. You can easily get a child position from the Canvas.
